src/drivers/intel/cavs/ipc.c

```diff
@@ -36,6 +36,15 @@ DECLARE_SOF_UUID("ipc-task", ipc_task_uuid, 0x8fa1d42f, 0xbc6f, 0x464b,
 
 /* No private data for IPC */
 
+
+#if CONFIG_IPC_MAJOR_4
+#if CAVS_VERSION < CAVS_VERSION_1_8
+#error "Invalid configuration for IPC4."
+#endif
+#define IPC4_EARLY_ACK 1   /* ack with DONE immediately */
```

Comment:
Basically with this you made sure that IPC4_EARLY_ACK is only defined in case CONFIG_IPC_MAJOR_4 and when CONFIG_IPC_MAJOR_4 is set the IPC4_EARLY_ACK is also. You can just use the CONFIG_IPC_MAJOR_4 in place of IPC4_EARLY_ACK

Reply:
True, but I kind of prefer to have more meaning in the ifdefs. It makes the code body more readable. Well, avoiding ifdefs to begin with would be even better, but if you have to do it, having some semantics seems helpful. But yeah, style issue.